scripts/lib-carousel.js

```diff
@@ -0,0 +1,389 @@
+  + '</svg>\n';
+
+let activeSlide = 1;
+let slideShow = false;
+let slideCount = 0;
+let touchStartX = 0;
+let touchEndX = 0;
+let touchRelX = 0;
+
+/**
+ * Increment active slide value
+ * @param direction
+ */
+const incrementActiveSlide = (direction = 1) => {
+  if (direction > 0) {
+    activeSlide = (activeSlide + 1) <= slideCount ? activeSlide + 1 : 1;
+  } else {
+    activeSlide = (activeSlide - 1) > 0 ? activeSlide - 1 : slideCount;
+  }
+  return activeSlide;
+};
+
+/**
+ * add attributes stored in key-value object to an element
+ * @param el
+ * @param attributes
+ */
+const addAttributes = (el, attributes) => {
+  Object.keys(attributes).forEach((key) => {
+    el.setAttribute(key, attributes[key]);
+  });
+};
+
+/**
+ * Activate/deactivate bottom bullets based on active slide
+ * @param bulletId
+ */
+const activateBullet = (bulletId) => {
+  const btnNav = document.querySelector('.dotted-nav');
+  [...btnNav.children].forEach((el) => {
+    const isActive = el.id === bulletId;
+    el.classList.toggle('active', isActive);
+    el.classList.toggle('inactive', !isActive);
+    el.setAttribute('aria-current', isActive ? 'true' : 'false');
+  });
+};
+
+/**
+ * Slides picture switcher
+ * @param slideId: format: slider-slide-${activeSlide}
+ */
+const activateSlide = (slideId) => {
+  const slider = document.querySelector('.slider');
+  Array.from(slider.children).forEach((el) => {
+    if (el.id === slideId) {
+      el.classList.replace('hide', 'show');
+      el.style.removeProperty('left');
+    } else {
+      el.classList.replace('show', 'hide');
+    }
+  });
+};
+
+/**
+ * Navigate with Bottom Bullet navigation
+ * @param event
+ */
+const dottedNavigation = (event) => {
+  const targetId = event.target.id.match(/\d+/)?.[0] || 1;
+  const sliderTarget = `slider-slide-${targetId}`;
+  activateSlide(sliderTarget);
+  activateBullet(event.target.id);
+  activeSlide = incrementActiveSlide();
+};
+
+/**
+ * Navigate with left and right for arrow navigation
+ * @param event
+ */
+const arrowNavigation = (event) => {
+  const navButton = event.currentTarget.id;
+  const increment = navButton === 'slider-arrow-left' ? -1 : 1;
+
+  activeSlide = incrementActiveSlide(increment);
+  activateSlide(`slider-slide-${activeSlide}`);
+  activateBullet(`slider-dot-${activeSlide}`);
+};
+
+/**
+ * Add event listeners for bottom bullet navigation
+ */
+const dottedNavOnClickEvents = () => {
+  const btnNav = document.querySelector('.dotted-nav');
+  btnNav.addEventListener('click', (event) => {
+    dottedNavigation(event);
+  });
+};
+
+/**
+ * Event Listeners for arrow navigation
+ */
+const arrowNavOnClickEvents = () => {
```

Comment:
Use cache and do event delegation:
```
const arrowNavOnClickEvents = () => {
  arrowNav.addEventListener('click', (event) => {
    if (event.target && event.target.nodeName === 'BUTTON') {
      arrowNavigation(event);
    }
  });
};
```

Reply:
For dottedNav it works but not for arrowNav